### PR TITLE
Diegesis/dev

### PR DESCRIPTION
**Pull request by hardikpoudeldev**
Task: https://github.com/etenlab/docs/issues/203

### CODE REVIEW

File: apps/diegesis-server/src/lib/makeServerHelpers/makeServerApp.js
Comment (mvahowe) [inline apps/diegesis-server/src/lib/makeServerHelpers/makeServerApp.js:26]: We can't (or at least shouldn't) do this in production. Anything that turns of security features needs to be inside the "debug" option, just below, and I think that already does what you need. In other words, you just need debug: true in your config file.

**Review by mvahowe — CHANGES_REQUESTED** [review]
The styling looks great, but a lot of the previous functionality no longer works. So, eg,

I think the footer text is hardwired. We need to get the text via GraphQL so that this text can be modified. (That was the reason for working from the mvh/dev branch which adds this functionality.)

Same for the home and other pages. GraphQL gives us the menu order plus the menu text and, in the case of static pages, the static page content.

The homepage may need some hardwired content. We need to decide how much of the homepage display is going to be hardwired into the component. But, eg, there's a "browse content" button which needs to either work or be removed. The entries/languages counts need to be driven by the data on the site. Etc.

In the list view, the entries don't open to show additional detail. The search features don't work either.

On the 'details' page the "downloads" button doesn't work, neither does the "select book" menu at the bottom of the page.

Basically, we can't merge this PR until everything that worked previously works again. Then, after merging this, we can get into how to configure styles, titles, logos etc, but the functionality has to come first.

Happy to chat about how best to make this happen.

File: apps/diegesis-server/src/lib/makeServerHelpers/makeServerApp.js
Comment (hardikpoudeldev) [reply apps/diegesis-server/src/lib/makeServerHelpers/makeServerApp.js:26]: yes, I agree I have updated it intentionally to work with different ports and It should not be part of the PR will fix it.

**Comment by hardikpoudeldev** [comment]
yes, I have seen changes in mvh/dev branch and it's updating the header, footer, and side nav menu as per GraphQL but after implementing Diegesis components it will allow you to set content in the diegesis component by passing props so it's mean I can't directly use markdown to render the content it would be nice if we receive GraphQL data in props form.

for other pages some of the functionalities not working as per the previous implementation and it's a little bit difficult to understand existing code and write it for the new design most of the functionality I covered. most probably I need some help understanding existing functionalities so that I can implement the same with the new design.

I'm planning to pick one page at a time and create a branch for that page and write functionality as per the new design and then raise the PR for the same.

The older design has more controls and functionality but I feel in the new design some of the controls are not available will discuss them once start working on each page.

File: apps/diegesis-server/src/lib/makeServerHelpers/makeServerApp.js
Comment (mvahowe) [reply apps/diegesis-server/src/lib/makeServerHelpers/makeServerApp.js:26]: You can set ports in the config file. I've tried to make pretty much everything configurable. If I missed something, let's see if we can add it at the config level, rather than changing the codebase.

**Comment by mvahowe** [comment]
> yes, I have seen changes in mvh/dev branch and it's updating the header, footer, and side nav menu as per GraphQL but after implementing Diegesis components it will allow you to set content in the diegesis component by passing props so it's mean I can't directly use markdown to render the content it would be nice if we receive GraphQL data in props form.

Why can't we pass a Markdown string to a component? If we don't do that, and if people want ever more variety in what appears in the footer, I think we end up reinventing our own version of Markdown anyway.

> for other pages some of the functionalities not working as per the previous implementation and it's a little bit difficult to understand existing code and write it for the new design most of the functionality I covered. most probably I need some help understanding existing functionalities so that I can implement the same with the new design.

Happy to help. The existing code is not very modular. We do need to decide whether it will be easier to bring the components into the project so that they can assume the Diegesis workflow. I don't have a firm opinion either way, but in some cases I think that the complexity of properties for a reusable component will make the code more complicated than just rewriting a component to do one thing, inline.

> I'm planning to pick one page at a time and create a branch for that page and write functionality as per the new design and then raise the PR for the same.

That sounds good. So shall we close this PR?

**Comment by hardikpoudeldev** [comment]
> > yes, I have seen changes in mvh/dev branch and it's updating the header, footer, and side nav menu as per GraphQL but after implementing Diegesis components it will allow you to set content in the diegesis component by passing props so it's mean I can't directly use markdown to render the content it would be nice if we receive GraphQL data in props form.
> 
> Why can't we pass a Markdown string to a component? If we don't do that, and if people want ever more variety in what appears in the footer, I think we end up reinventing our own version of Markdown anyway.

For the markdown string as I can see ReactMarkdown react plugin will render the entire string as per the markdown syntax let's take an example of a footer section as per the diegesis component it has two parts on the left part it's showing the content of the footer and on the right side, it's showing branch name and also footer section responsive as per the mobile view then how to react markdown string going to handle these things. for better understanding I will recommend to check PageFooter component in diegesis ui-kit repo.

> > for other pages some of the functionalities not working as per the previous implementation and it's a little bit difficult to understand existing code and write it for the new design most of the functionality I covered. most probably I need some help understanding existing functionalities so that I can implement the same with the new design.
> 
> Happy to help. The existing code is not very modular. We do need to decide whether it will be easier to bring the components into the project so that they can assume the Diegesis workflow. I don't have a firm opinion either way, but in some cases I think that the complexity of properties for a reusable component will make the code more complicated than just rewriting a component to do one thing, inline.

It will be easier to bring the components into the project so that we can directly apply the props and add the functionality and it also helps us to make changes as per requirement. I know Diegesis ui-kit exporting both pages and its child components and on most of the pages, I'm directly using page components and passing all props.

> > I'm planning to pick one page at a time and create a branch for that page and write functionality as per the new design and then raise the PR for the same.
> 
> That sounds good. So shall we close this PR?

Yes, we can close this PR but I will create multiple PRs based on these current changes because most of the functionalities are working.

**Pull request closed without merging**
